apps/balancer-tools/src/app/poolsimulator/(components)/SwapSimulator.tsx

```diff
@@ -290,3 +143,293 @@ function calculateSimulation({
+    const tokenInSymbol = tokensSymbol?.[Number(data.tokenInIndex)];
+    const tokenOutSymbol = tokensSymbol?.[Number(data.tokenOutIndex)];
+    const initialResult = calculateSimulation({
+      amm: initialAMM,
+      swapType: data.swapType as string,
+      amount: Number(data.amount),
+      tokenInSymbol: tokenInSymbol,
+      tokenOutSymbol: tokenOutSymbol,
+    });
+    const customResult = calculateSimulation({
+      amm: customAMM,
+      swapType: data.swapType as string,
+      amount: Number(data.amount),
+      tokenInSymbol: tokenInSymbol,
+      tokenOutSymbol: tokenOutSymbol,
+    });
+    setInitialResult({
+      swapType: data.swapType as string,
+      tokenInSymbol: tokenInSymbol,
+      tokenOutSymbol: tokenOutSymbol,
+      ...initialResult,
+    });
+    setCustomResult({
+      swapType: data.swapType as string,
+      tokenInSymbol: tokenInSymbol,
+      tokenOutSymbol: tokenOutSymbol,
+      ...customResult,
+    });
+  };
+
+  useEffect(() => {
+    const defaultInitialResult = calculateSimulation({
+      amm: initialAMM,
+      swapType: defaultSwapType,
+      amount: Number(defaultAmount),
+      tokenInSymbol: tokensSymbol?.[Number(defaultTokenInIndex)],
+      tokenOutSymbol: tokensSymbol?.[Number(defaultTokenOutIndex)],
+    });
+    const defaultCustomResult = calculateSimulation({
+      amm: customAMM,
+      swapType: defaultSwapType,
+      amount: Number(defaultAmount),
+      tokenInSymbol: tokensSymbol?.[Number(defaultTokenInIndex)],
+      tokenOutSymbol: tokensSymbol?.[Number(defaultTokenOutIndex)],
+    });
+    setInitialResult({
+      swapType: defaultSwapType,
+      tokenInSymbol: tokensSymbol?.[Number(defaultTokenInIndex)],
+      tokenOutSymbol: tokensSymbol?.[Number(defaultTokenOutIndex)],
+      ...defaultInitialResult,
+    });
+    setCustomResult({
+      swapType: defaultSwapType,
+      tokenInSymbol: tokensSymbol?.[Number(defaultTokenInIndex)],
+      tokenOutSymbol: tokensSymbol?.[Number(defaultTokenOutIndex)],
+      ...defaultCustomResult,
+    });
+  }, [initialAMM, customAMM]);
+
+  return (
+    <Form {...form} onSubmit={onSubmit}>
+      <div className="grid grid-cols-2 gap-2">
+        <div>
+          <FormLabel className="mb-2 block text-sm text-slate12">
+            Token In
+          </FormLabel>
+          <Controller
+            control={control}
+            name="tokenInIndex"
+            defaultValue={defaultTokenInIndex}
+            render={({ field: { onChange, value, ref } }) => (
+              <Select
+                onValueChange={onChange}
+                value={value as string}
+                ref={ref}
+                className="w-full"
+              >
+                {tokensSymbol.map((tokenSymbol, index) => (
+                  <SelectItem key={tokenSymbol} value={index.toString()}>
+                    {tokenSymbol}
+                  </SelectItem>
+                ))}
+              </Select>
+            )}
+          />
```

Comment:
Could also be abstracted to avoid repeating almost the same thing three times here